airflow/www/views.py

```diff
@@ -365,7 +365,10 @@ def get_date_time_num_runs_dag_runs_form_data(request, session, dag):
     dr_choices = []
     dr_state = None
     for dr in drs:
-        dr_choices.append((dr.execution_date.isoformat(), dr.run_id))
+        # Parse the timestamp from the run id and manually convert it to EST for the UI
+        prefix = dr.run_id.split("__")[0]
+        label = prefix + "__" + timezone.convert_to_est(dr.execution_date).isoformat()
```

Comment:
use f-strings?
```suggestion
        label = f"{prefix}__{timezone.convert_to_est(dr.execution_date).isoformat()}"
```
or do a join to go with the split?
```suggestion
        label = "__".join([prefix, timezone.convert_to_est(dr.execution_date).isoformat()]
```